packages/@glimmer/runtime/lib/compiled/opcodes/component.ts

```diff
@@ -258,16 +279,31 @@ APPEND_OPCODES.add(Op.PrepareArgs, (vm, { op1: _state }) => {
   stack.push(args);
 });
 
+function resolveCurriedComponentDefinition(instance: ComponentInstance, definition: CurriedComponentDefinition, args: Arguments): ComponentDefinition {
+  let unwrappedDefinition = instance.definition = definition.unwrap(args);
+  let { manager, state } = unwrappedDefinition;
+
+  assert(instance.manager === null, "component instance manager should not be populated yet");
```

Comment:
Can't you just use `expect`

Reply:
Doesn't `expect` assert that something is *not* null?

Reply:
An `assert` is right here.

Reply:
@wycats Huh?

Reply:
I'm agreeing with you.